sklego/meta/_grouped_utils.py

```diff
@@ -88,7 +91,46 @@ def _check_grouping_columns(X_group, **kwargs) -> pd.DataFrame:
 
     # Only check missingness in object columns
     if X_group.select_dtypes(exclude="number").isnull().any(axis=None):
-        raise ValueError("X has NaN values")
+        raise ValueError("Group columns contain NaN values")
 
     # The grouping part we always want as a DataFrame with range index
     return X_group.reset_index(drop=True)
+
+
+def _get_estimator(estimators, grp_values, grp_names, return_level, fallback_method):
```

Comment:
The point of this function is to determine which estimator to use to predict.
- if `fallback_method ="raise"`, we have to have the model for the group we are predicting
- if `fallback_method ="next"`, we check for recursively for the first available parent
- if `fallback_method ="global"`, we summon the global model.

The point of returning a `return_level` is a trick to know how far back we went, and used to slice an array afterwards (more comments where this happens)